compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp

```diff
@@ -17,10 +17,31 @@
 #include "mlir/Pass/PassManager.h"
 #include "mlir/Transforms/Passes.h"
 
+/// Command line options used purely for development purposes. Not to be relied
+/// on in any way.
+static llvm::cl::opt<bool> clUseCPlusPlusTransformPasses(
+    "iree-amd-aie-cpp-passes",
+    llvm::cl::desc(
+        "Runs the cpp passes instead of transform dialect when possible"),
+    llvm::cl::init(true));
+
 namespace mlir::iree_compiler::AMDAIE {
 
 void buildAMDAIETransformPassPipeline(OpPassManager &pm) {
   addCommonTargetExecutablePreprocessingPasses(pm);
+  // TODO: Current we don't include C++ equivalent of Transform dialect scripts.
+  // We are thus guarding their inclusion with a bool
+  // `useCPlusPlusTransformPasses` which would be used during development
+  // efforts. Once the C++ passes are ready, we will include these passes by
+  // default and take away the guarding.
+  if (clUseCPlusPlusTransformPasses) {
+    pm.addPass(createAMDAIETileAndFusePass(1));
+    pm.addPass(createAMDAIEPadAndBufferizePass());
+    pm.addPass(createCleanupPass());
+    pm.addPass(createCSEPass());
+    // pm.addPass(createAMDAIETileAndFusePass(2));
+  }
```

Comment:
Should we reorganize `buildAMDAIETransformPassPipeline` and add padding pass in a way similar to LLVMCPU pipelines? https://github.com/openxla/iree/blob/73f5947ff170fbf1ea8e415c019673cb7dd06aea/compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp#L280

Reply:
Fair ask. Although, for similar justification I'd request taking these reorganisation after we have an e2e C++ for at least the Matmul Pad flow. Would help better once we have the complete picture ready.